packages/block-library/src/accordion/view.js

```diff
@@ -1,38 +1,87 @@
 /**
  * WordPress dependencies
  */
-import { store, getContext } from '@wordpress/interactivity';
+import { store, getContext, withSyncEvent } from '@wordpress/interactivity';
 
-const { state } = store( 'core/accordion', {
+store( 'core/accordion', {
 	state: {
 		get isOpen() {
-			const { isOpen, id } = getContext();
-			return isOpen.includes( id );
+			const { id, accordionContents } = getContext();
+			const accordionContent = accordionContents.find(
+				( item ) => item.id === id
+			);
+			return accordionContent ? accordionContent.isOpen : false;
 		},
 	},
 	actions: {
 		toggle: () => {
 			const context = getContext();
-			const { id, autoclose } = context;
+			const { id, autoclose, accordionContents } = context;
+			const accordionContent = accordionContents.find(
+				( item ) => item.id === id
+			);
 
 			if ( autoclose ) {
-				context.isOpen = state.isOpen ? [] : [ id ];
-			} else if ( state.isOpen ) {
-				context.isOpen = context.isOpen.filter(
-					( item ) => item !== id
-				);
+				accordionContents.forEach( ( item ) => {
+					item.isOpen =
+						item.id === id ? ! accordionContent.isOpen : false;
+				} );
 			} else {
-				context.isOpen.push( id );
+				accordionContent.isOpen = ! accordionContent.isOpen;
 			}
 		},
+		handleKeyDown: withSyncEvent( ( event ) => {
```

Comment:
Unfortunately, the `withSyncEvent` wrapper doesn't seem to work as expected. If you press arrowUp/ArrowDown/Home/End key on the toggle button, the browser console logs the following warning:

<img width="696" height="98" alt="Image" src="https://github.com/user-attachments/assets/7c73b36c-9bca-471e-a859-d6894d7d8ca4" />

I believe this problem is the same as #69975, and should be fixed in #70325.